src/app-layout/__tests__/trigger-button.test.tsx

```diff
@@ -0,0 +1,353 @@
+        expect(button).toBeTruthy();
+        expect(button!.getElement().getAttribute('aria-label')).toEqual(mockProps.ariaLabel);
+        expect(wrapper!.findIcon()).toBeTruthy();
+        expect(wrapper.findByClassName(visualRefreshStyles.dot)).toBeNull();
+        expect(wrapper.findBadge()).toBeNull();
+      });
+
+      test('renders correctly with aria controls', () => {
+        const ref: React.MutableRefObject<ButtonProps.Ref | null> = React.createRef();
+        const mockAriaControls = 'mock-aria-control';
+        const { wrapper, getByTestId } = renderVisualRefreshTriggerButton(
+          {
+            badge: false,
+            ariaControls: mockAriaControls,
+          },
+          {
+            isMobile,
+            hasOpenDrawer,
+          },
+          ref
+        );
+
+        expect(wrapper).not.toBeNull();
+        const button = wrapper.find('button');
+        expect(getByTestId(mockTestId)).toBeTruthy();
+        expect(button).toBeTruthy();
+        expect(button!.getElement().getAttribute('aria-label')).toEqual(mockProps.ariaLabel);
+        expect(button!.getElement().getAttribute('aria-controls')).toEqual(mockAriaControls);
+      });
+
+      test('renders correctly with iconName', () => {
+        const ref: React.MutableRefObject<ButtonProps.Ref | null> = React.createRef();
+        const { wrapper, getByTestId } = renderVisualRefreshTriggerButton({}, {}, ref);
+
+        expect(wrapper).not.toBeNull();
+        const button = wrapper.find('button');
+        expect(getByTestId(mockTestId)).toBeTruthy();
+        expect(button).toBeTruthy();
+        expect(button!.getElement().getAttribute('aria-label')).toEqual(mockProps.ariaLabel);
+        expect(wrapper!.findIcon()).toBeTruthy();
+      });
+
+      test('renders correctly with iconSvg', () => {
+        const iconTestId = 'icon-test-id';
+        const icon = (
+          <svg data-testid={iconTestId} viewBox="0 0 24 24">
+            <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z" />
+          </svg>
+        );
+        const ref: React.MutableRefObject<ButtonProps.Ref | null> = React.createRef();
+        const { wrapper, getByTestId } = renderVisualRefreshTriggerButton(
+          {
+            iconSvg: icon,
+          },
+          {
+            isMobile,
+            hasOpenDrawer,
+          },
+          ref
+        );
+
+        expect(wrapper).not.toBeNull();
+        const button = wrapper.find('button');
+        expect(getByTestId(mockTestId)).toBeTruthy();
+        expect(button).toBeTruthy();
+        expect(button!.getElement().getAttribute('aria-label')).toEqual(mockProps.ariaLabel);
+        expect(wrapper!.findIcon()).toBeTruthy();
+        expect(getByTestId(iconTestId)).toBeTruthy();
+      });
+
+      test.each([true, false])('renders correctly and with when disabled is %s', disabledValue => {
```

Comment:
The description is unclear: with what it renders when disabled?

Reply:
nit: I would avoid saying "renders correctly" because that is vague. It is better to express the clear expectation like "aria label is assigned" or "disabled reasons is shown" etc.